src/main/java/com/google/sps/servlets/UserTripServlet.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import com.google.sps.Trip;
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/user-trips")
+public class UserTripServlet extends HttpServlet {
+
+  // This allows the doGet method to return null.
+  private static final String nullReturn = null;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+
+    // If no user is signed in, return null.
+    Entity userEntity = AuthServlet.getCurrentUserEntity();
+    if (userEntity == null) {
+      response.getWriter().println(nullReturn);
+      return;
+    }
+
+    // Get the list of Trip Entity objects through a Query.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Has Adam talked to you about creating utility methods for these DatastoreService calls? These seem to be repeating a lot throughout the codebase, and I think it might be a good idea to create methods that make these calls so you, Eshika, and Adam don't all have to write the same calls with slight differences.

Not only will it make it easier to read/write, but will also make the code stronger, as whenever you want to change how these calls are made or how the elements are stored, you'll only have to make changes in one place instead of 3.

I'm not necessarily going to make you write these changes now, but I really want you three to chat about this and come up with a plan forward

Reply:
Yes, Adam mentioned it! We (Eshika, Adam, and I) still have to hop on a call to talk about this, and can hopefully do so late today or tomorrow morning.